script/CREATE3Script.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: BSD 3-Clause License
+pragma solidity ^0.8.24;
+
+import {ICREATE3Factory} from "@create3-factory/ICREATE3Factory.sol";
+import {TransparentUpgradeableProxy} from
+    "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+import {L1YnOFTAdapterUpgradeable} from "@/L1YnOFTAdapterUpgradeable.sol";
+import {L2YnERC20Upgradeable} from "@/L2YnERC20Upgradeable.sol";
+import {L2YnOFTAdapterUpgradeable} from "@/L2YnOFTAdapterUpgradeable.sol";
+
+// @dev This contract is used to deploy contracts and proxies using CREATE3.
+//      It should be inherited by the script that deploys the contracts, it shouldn't be used directly or as a
+// library.
+abstract contract CREATE3Script {
+    ICREATE3Factory public constant CREATE3_FACTORY = ICREATE3Factory(0x3Ab34A5758F42080A536865aD3a7D35E92861418);
+
+    function deployProxy(
```

Comment:
all helper functions from `ImmutableMultiChainDeployer` has been moved here, they use CREATE3Factory deploy the contracts now